scipy/sparse/linalg/_eigen/_svds.py

```diff
@@ -338,49 +323,28 @@ def matmat_XH_X(x):
     elif solver == 'arpack' or solver is None:
         if v0 is None and not rs_was_None:
             v0 = random_state.uniform(size=(min(A.shape),))
-        eigvals, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
+        _, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
                                 ncv=ncv, which=which, v0=v0)
 
-    # Gramian matrices have real non-negative eigenvalues.
-    eigvals = np.maximum(eigvals.real, 0)
-
-    # Use the sophisticated detection of small eigenvalues from pinvh.
-    t = eigvec.dtype.char.lower()
-    factor = {'f': 1E3, 'd': 1E6}
-    cond = factor[t] * np.finfo(t).eps
-    cutoff = cond * np.max(eigvals)
-
-    # Get a mask indicating which eigenpairs are not degenerately tiny,
-    # and create the re-ordered array of thresholded singular values.
-    above_cutoff = (eigvals > cutoff)
-    nlarge = above_cutoff.sum()
-    nsmall = k - nlarge
-    slarge = np.sqrt(eigvals[above_cutoff])
-    s = np.zeros_like(eigvals)
-    s[:nlarge] = slarge
+    eigvec, _ = np.linalg.qr(eigvec)
```

Comment:
Are we not happy with what ARPACK or LOBPCG eigenvectors here and renormalizing it? 

This might actually introduce some numerical problems or even exaggerate when eigenvectors are not separated properly due to heavy reflector arithmetics.

Reply:
Yes, but not just renormalizing - reorthogonalizing, for safety. Please see comment in line 771 of https://github.com/scipy/scipy/blob/main/scipy/sparse/linalg/_eigen/lobpcg/lobpcg.py

No, to my knowledge the QR here does not introduce any numerical problems other than some extra costs, as is well tested in the new unit tests. If you know an example where the QR here makes the computations less robust, please share.

Reply:
This would close #12629. What about having a `cutoff` parameter otherwise?